src/endpoint/rules.ts

```diff
@@ -1,174 +1,168 @@
 	aggregation?: ConditionAggregationMode
 }
 
 export interface BetweenCondition {
-	/**
-	 * Unique id for the condition
-	 */
-	id?: string
-	value: Operand
-	start: Operand
-	end: Operand
+	value: RuleOperand
+	start: RuleOperand
+	end: RuleOperand
 	aggregation?: ConditionAggregationMode
 }
 
-export interface SingleOperandCondition {
-	/**
-	 * Unique id for the condition
-	 */
-	id?: string
-	left: Operand
-	aggregation?: ConditionAggregationMode
-}
-
-export interface TimedSimpleCondition {
-	id?: string
-	left: Operand
-	right: Operand
-	interval: Interval
-	aggregation?: ConditionAggregationMode
+/**
+ * A condition that returns true when its evaluation resolves to true and the previous evaluation
+ * resolved to false.
+ */
+export interface ChangesCondition extends BasicCondition {
+	id: string
+	operand?: RuleOperand
 }
 
-export enum ArmState {
-	ArmedStay = 'ArmedStay',
-	ArmedAway = 'ArmedAway',
-	Disarmed = 'Disarmed',
+/**
+ * A condition that returns true if its evaluation would return true within the specified duration.
+ */
+export interface WasCondition extends BasicCondition {
+	id: string
+	operand?: RuleOperand
+	duration: RuleInterval
 }
 
-export interface SecurityState {
-	armState: ArmState
-	monitoring?: boolean
+/**
+ * A condition that returns true if its evaluation is true after the specified duration.
+ */
+export interface RemainsCondition extends BasicCondition {
+	id: string
+	duration: RuleInterval
+	operand?: RuleOperand
 }
 
-export interface Condition {
-	and?: Condition[]
-	or?: Condition[]
-	not?: Condition
+export interface BasicCondition {
+	and?: RuleCondition[]
+	or?: RuleCondition[]
+	not?: RuleCondition
 	equals?: SimpleCondition
 	greaterThan?: SimpleCondition
 	greaterThanOrEquals?: SimpleCondition
 	lessThan?: SimpleCondition
 	lessThanOrEquals?: SimpleCondition
 	between?: BetweenCondition
-	dropsBelow?: SimpleCondition
-	dropsToOrBelow?: SimpleCondition
-	risesAbove?: SimpleCondition
-	risesToOrAbove?: SimpleCondition
 }
 
-export enum SubscriptionMode {
-	Auto = 'Auto',
-	Always = 'Always',
-	Never = 'Never'
-}
-
-export enum TimeReference {
-	Now = 'Now',
-	Midnight = 'Midnight',
-	Sunrise = 'Sunrise',
-	Noon = 'Noon',
-	Sunset = 'Sunset'
-}
-
-export interface TimeOperand {
-	/**
-	 * A java time zone ID reference
-	 */
-	timeZoneId?: string
-	reference: TimeReference
-	offset?: Interval
+export interface RuleCondition {
+	and?: RuleCondition[]
+	or?: RuleCondition[]
+	not?: RuleCondition
+	equals?: SimpleCondition
+	greaterThan?: SimpleCondition
+	greaterThanOrEquals?: SimpleCondition
+	lessThan?: SimpleCondition
+	lessThanOrEquals?: SimpleCondition
+	between?: BetweenCondition
+	changes?: ChangesCondition
+	remains?: RemainsCondition
+	was?: WasCondition
 }
 
-export enum OperandAggregationMode {
-	None = 'None',
-	Avg = 'Avg',
-	Sum = 'Sum',
-	Min = 'Min',
-	Max = 'Max',
-	Least = 'Least'
-}
+export type TimeReference ='Now' | 'Midnight' | 'Sunrise' | 'Noon' | 'Sunset'
```

Comment:
Nit: space after `=`

Here is an eslint rule we could add that covers this: [@typescript-eslint/space-infix-ops](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/space-infix-ops.md#how-to-use).

Reply:
Fixed and I added that rule.